### PR TITLE
Add table rdap_domain (#36)

### PR DESCRIPTION
**Pull request by ParthaI**
# Example query results
<details>
  <summary>Results</summary>
  
```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

File: whois/table_rdap_domain.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			Hydrate:    getRdapDomain,
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "domain", Type: proto.ColumnType_STRING, Transform: transform.FromQual("domain"), Description: "Domain name the RDAP information relates to."},
+
+			{Name: "domain_id", Type: proto.ColumnType_STRING, Transform: transform.FromField("Handle").NullIfZero(), Description: "Unique identifier for the domain."},
+			{Name: "object_class_name", Type: proto.ColumnType_STRING, Description: "String 'domain' representing the object type in RDAP."},
+			{Name: "ldh_name", Type: proto.ColumnType_STRING, Description: "Textual representation of DNS names.", Transform: transform.FromField("LDHName")},
+			{Name: "port43", Type: proto.ColumnType_STRING, Description: "Hostname of Registry WHOIS server."},
```
Comment (cbruno10) [inline whois/table_rdap_domain.go:28]: The description seems a bit strange based on the column name, do you have an example of the data we get back from the SDK for this field/column?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@ParthaI Please see comment, thanks!

File: whois/table_rdap_domain.go (hunk 1 above)
Comment (ParthaI) [reply whois/table_rdap_domain.go:28]: @cbruno10, I queried this table  to retrieve the 'port43' value by using multiple domain names. However, in all cases, we received null or empty data.

I found the description mentioned in the [document](https://fred.nic.cz/documentation/html/RDAPReference/Domain.html), so I included it here.

```
> select domain, port43 from rdap_domain where domain = 'example.com'
+-------------+--------+
| domain      | port43 |
+-------------+--------+
| example.com |        |
+-------------+--------+

Time: 3.2s. Rows fetched: 1. Hydrate calls: 0.
> select domain, port43 from rdap_domain where domain = 'google.org'
+------------+--------+
| domain     | port43 |
+------------+--------+
| google.org |        |
+------------+--------+

Time: 4.5s. Rows fetched: 1. Hydrate calls: 0.
> select domain, port43 from rdap_domain where domain = 'microsoft.info'
+----------------+--------+
| domain         | port43 |
+----------------+--------+
| microsoft.info |        |
+----------------+--------+

Time: 2.2s. Rows fetched: 1. Hydrate calls: 0.
> select domain, port43 from rdap_domain where domain = 'whitehouse.gov'
+--------+--------+
| domain | port43 |
+--------+--------+
+--------+--------+

Time: 354ms.
> select domain, port43 from rdap_domain where domain = 'amazon.in'
+--------+--------+
| domain | port43 |
+--------+--------+
+--------+--------+

Time: 399ms.
> select domain, port43 from rdap_domain where domain = 'amazon.com'
+------------+--------+
| domain     | port43 |
+------------+--------+
| amazon.com |        |
+------------+--------+

Time: 1.2s. Rows fetched: 1. Hydrate calls: 0.
```

File: whois/table_rdap_domain.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		Description: "RDAP domain information including expiration, DNS servers and contact details.",
+		List: &plugin.ListConfig{
+			KeyColumns: plugin.SingleColumn("domain"),
+			Hydrate:    getRdapDomain,
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "domain", Type: proto.ColumnType_STRING, Transform: transform.FromQual("domain"), Description: "Domain name the RDAP information relates to."},
+
+			{Name: "domain_id", Type: proto.ColumnType_STRING, Transform: transform.FromField("Handle").NullIfZero(), Description: "Unique identifier for the domain."},
```
Comment (cbruno10) [inline whois/table_rdap_domain.go:25]: Should the column be called `handle`, if the field name is `Handle`? Is there a reason you chose `domain_id` instead? Also, where does the description come from?

File: docs/tables/rdap_domain.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
+# Table: rdap_domain
```
Comment (cbruno10) [inline docs/tables/rdap_domain.md:1]: @ParthaI Can you please update this table doc so it matches our latest doc format (if it doesn't already)?

**Review by cbruno10 — COMMENTED** [review]
@ParthaI Please see comments, thanks!

File: whois/table_rdap_domain.go (hunk 2 above)
Comment (ParthaI) [reply whois/table_rdap_domain.go:25]: @cbruno10, I've renamed the column domain_id to handle to ensure consistency with the API response. For most column descriptions, I referred to [this documentation](https://fred.nic.cz/documentation/html/RDAPReference/Domain.html). For those not covered in the documentation, I searched online and selected descriptions that best matched the data contained in each column.

https://datatracker.ietf.org/doc/html/rfc7483.html#section-5.1
https://datatracker.ietf.org/doc/html/rfc7483#section-5.3

File: docs/tables/rdap_domain.md (hunk 3 above)
Comment (ParthaI) [reply docs/tables/rdap_domain.md]: @cbruno10, I have updated the dos to match our latest doc format.

File: docs/tables/rdap_domain.md [hunk 4]
```diff
@@ -0,0 +1,374 @@
+---
+title: "Steampipe Table: rdap_domain - Query Whois Domains using SQL"
```
Comment (cbruno10) [inline docs/tables/rdap_domain.md:2]:
```suggestion
title: "Steampipe Table: rdap_domain - Query RDAP Domains using SQL"
```

File: docs/tables/rdap_domain.md [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+
+The `rdap_domain` table provides insights about an RDAP domain query specifically refers to querying information about a domain name. This can include details such as the domain name's registration status, the registrar information, the domain's creation and expiration dates, and contact information associated with the domain. RDAP provides a more secure and standardized way to access this information compared to WHOIS, and it is becoming the preferred method for domain-related queries in the internet infrastructure community.
+
+**Important Notes**
+It's not practical to list all domains in the world, so this table requires a
+`domain` qualifier to be passed in the `where` or `join` clause for all queries.
+
+## Examples
+
+### Basic whois info
```
Comment (cbruno10) [inline docs/tables/rdap_domain.md:20]:
```suggestion
### Basic RDAP info
```

File: whois/plugin.go [hunk 6]
```diff
@@ -13,6 +13,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 		DefaultTransform: transform.FromGo().NullIfZero(),
 		TableMap: map[string]*plugin.Table{
 			"whois_domain": tableWhoisDomain(ctx),
+			"rdap_domain":  tableRdapDomain(ctx),
```
Comment (cbruno10) [inline whois/plugin.go:16]: Please move above so we keep alphabetical order

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@ParthaI Please see review comments, thanks!

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**